osquery/tables/system/tests/posix/augeas_tests.cpp

```diff
@@ -87,5 +87,56 @@ TEST_F(AugeasTests, select_hosts_by_node) {
       << "Value is not empty. Got " << results.rows()[0].at("value")
       << "instead";
 }
+
+TEST_F(AugeasTests, select_augeas_load) {
+  auto results = SQL("select * from augeas where node = '/augeas/load'");
+  ASSERT_EQ(results.rows().size(), 1U);
+  ASSERT_EQ(results.rows()[0].at("node"), "/augeas/load");
+  ASSERT_EQ(results.rows()[0].at("label"), "load");
+  ASSERT_TRUE(results.rows()[0].at("path").empty());
+  ASSERT_TRUE(results.rows()[0].at("value").empty());
+}
+
+TEST_F(AugeasTests, select_augeas_load_wildcards) {
+  // Exact matches, should be 1 result
+  ASSERT_EQ(
+      SQL("select * from augeas where node LIKE '/augeas/load'").rows().size(),
+      1U);
+  ASSERT_EQ(SQL("select * from augeas where node LIKE '/%/load'").rows().size(),
+            1U);
+
+  // Single recurse, about 200 results
+  ASSERT_GT(SQL("select * from augeas where node LIKE '/augeas/load/%'")
+                .rows()
+                .size(),
+            100U);
+
+  // full recuse, about 1500 results
+  ASSERT_GT(SQL("select * from augeas where node LIKE '/augeas/load/%%'")
+                .rows()
+                .size(),
+            1000U);
+}
+
+TEST_F(AugeasTests, select_file_wildcards) {
+  // These are a bit funny. Augeas doesn't do partial matches,
+  // and because file is a real file, you have to be carefuly
+  // with trailing slashes.
+  ASSERT_EQ(
+      SQL("select * from augeas where path LIKE '/etc/hosts/%'").rows().size(),
+      0U);
+  ASSERT_EQ(
+      SQL("select * from augeas where path LIKE '/etc/hosts%'").rows().size(),
```

Comment:
It seems odd that this returns 0 results. It sounds like this is an augeas quirk vs. a confound the SQL abstraction adds?

Reply:
Yes, correct. AFAICT augeas doesn't do partial matches on a leaf's name:
```
augtool> match /files/etc/shells/1
/files/etc/shells/1 = /bin/bash

augtool> match /files/etc/shells/1*
  (no matches)

augtool> match /files/etc/shells/*
/files/etc/shells/#comment[1] = List of acceptable shells for chpass(1).
/files/etc/shells/#comment[2] = Ftpd will not allow users to connect who are not using
/files/etc/shells/#comment[3] = one of these shells.
/files/etc/shells/1 = /bin/bash
/files/etc/shells/2 = /bin/csh
/files/etc/shells/3 = /bin/ksh
/files/etc/shells/4 = /bin/sh
/files/etc/shells/5 = /bin/tcsh
/files/etc/shells/6 = /bin/zsh
```

Thinking aloud... I went down this path because I couldn't query `LIKE` on the `/augeas` hierarchy. This is because the current implementation has complexity to pass `EQUALS` around. But when there are no `EQUALS`, it defaults to matching off `/files//*` and letting sqlite filter. Which obviously omits the `/augeas/`/ tree.

This approach extends the prior logic parsing the sql expressions to include `LIKE`. And barring a lot more complexity in the translation, I think this means it's subject to the underlying augtool limitations. 

I think there are other possible implementations.

We could change the default match to `//*`, which would dump the entire augeas tree back to sqlite for filtering. This feels simple. As I think the entire filespace is loaded into augeas on initialization, I don't think there'd be a significant performance impact. It doesn't feel right, but I'm not sure why. 

We could change the general handling to _not_ auto load files unless explicitly requested, then return the full space and let sqlite filter. This is appealing to me, but I don't see how to do it performantly. 

Though, one advantage to the current approach is that you can use the `%` vs `%%` difference to explore a path. For example:
```
osquery> select node from augeas where node like '/augeas/%';
+-------------------+
| node              |
+-------------------+
| /augeas/root      |
| /augeas/context   |
| /augeas/variables |
| /augeas/version   |
| /augeas/save      |
| /augeas/span      |
| /augeas/load      |
| /augeas/files     |
+-------------------+
osquery> select node from augeas where node like '/augeas/files/%';
+---------------------+
| node                |
+---------------------+
| /augeas/files/Users |
| /augeas/files/etc   |
+---------------------+
```